include/nbl/builtin/hlsl/emulated/vector_t.hlsl

```diff
@@ -183,6 +175,14 @@ NBL_CONSTEXPR_FUNC vector<bool, CRTP::Dimension> operator OP (vector<component_t
 #define NBL_EMULATED_VECTOR_CREATION_AND_COMPONENT_SUM \
 using this_t = emulated_vector<ComponentType, CRTP>;\
 using component_t = ComponentType;\
+template<typename T>\
+NBL_CONSTEXPR_STATIC this_t create(vector<T, CRTP::Dimension> other)\
```

Comment:
this needs `NBL_FUNC_REQUIRES(impl::assignable_v<component_t,T>)` or something to check that you can assign